### PR TITLE
Se 453 refactor all amm calculators to follow nft calculator pattern

### PR DESCRIPTION
**Pull request by oscarsernarosero**

### CODE REVIEW

File: src/liquidity/calculators/ProtocolAMMCalcLinear.sol [hunk 1]
```diff
@@ -2,157 +2,89 @@
...
-            return _amount1;
-        } else {
-            // swap token1 for token0
-            _amount0 =
-                ((2 * (10 ** 9)) * (_amount1 * b_denom) * sqrt(m_denom)) /
-                (sqrt(((10 ** 18) * (b_num ** 2) * m_denom) + 2 * _reserve1 * m * (b_denom ** 2)) + sqrt(((10 ** 18) * (b_num ** 2) * m_denom) + 2 * (_reserve1 - _amount1) * m * (b_denom ** 2)));
-
-            return _amount0;
-        }
+        return simulateSwap( _reserve0, _reserve1, _amount0, _amount1);
```
Comment (ShaneDuncan602) [inline src/liquidity/calculators/ProtocolAMMCalcLinear.sol:47]: I like how these funnel through simulateSwap.

File: src/liquidity/calculators/libraries/Curve.sol [hunk 2]
```diff
@@ -12,65 +12,171 @@ import {Math} from "./Math.sol";
...
+            uint256 trackerSquareRoot = uint(curve.tracker).sqrt();
+            if(deltaYSquareRoot < trackerSquareRoot) 
+                revert InsufficientPoolDepth();
+            amountOut = (10 ** 9) * (deltaYSquareRoot - trackerSquareRoot); /// check for tracker sign!!
+        
+        } else {
+            uint256 tenMinusTrackerSquare = uint((((10 ** 19)) - curve.tracker) ** 2);
+            uint256 tenMinusDeltaSquare = uint(((10 ** 19) - ((curve.tracker + int(curve.amountIn)))) ** 2);
+            if(tenMinusTrackerSquare < tenMinusDeltaSquare) 
+                revert InsufficientPoolDepth();
```
Comment (ShaneDuncan602) [inline src/liquidity/calculators/libraries/Curve.sol:177]: Shouldn't we just check to make sure that the amount going out is not greater than the reserve of that token? That could just be done in the AMM itself. Nevermind, I see that check is in the AMM already. These checks should be run by Neel during the meeting.

**Review by ShaneDuncan602 — APPROVED** [review]
This looks very good, Oscar. This was a complext ticket and you killed it. I'm approving. We will likely have changes after we get more clarification from Neel, but I think it's ready.

**Review by TJ-Everett — APPROVED** [review]
Amazing work Oscar, this looks great!

**Pull request merged**